system/CodeIgniter.php

```diff
@@ -1063,14 +1063,18 @@ public function storePreviousURL($uri)
             $uri = new URI($uri);
         }
 
-        if (isset($_SESSION)) {
-            $_SESSION['_ci_previous_url'] = URI::createURIString(
-                $uri->getScheme(),
-                $uri->getAuthority(),
-                $uri->getPath(),
-                $uri->getQuery(),
-                $uri->getFragment()
-            );
+        $previousUrl = URI::createURIString(
+            $uri->getScheme(),
+            $uri->getAuthority(),
+            $uri->getPath(),
+            $uri->getQuery(),
+            $uri->getFragment()
+        );
+
+        if (ENVIRONMENT === 'testing') {
```

Comment:
Eventually this condition (`ENVIRONMENT === 'testing'`) should be removed in all the code base.
And the Session class should work in testing.
Is there something wrong?

Reply:
If you don't have this validation, there are a series of tests that fail loke this:

```
ErrorException : ini_set(): Headers already sent. You cannot change the session module's ini settings at this time
 D:\projects\CodeIgniter4\system\Session\Handlers\FileHandler.php:72
 D:\projects\CodeIgniter4\system\Config\Services.php:686
 D:\projects\CodeIgniter4\system\Config\BaseService.php:258
 D:\projects\CodeIgniter4\system\Config\BaseService.php:199
 D:\projects\CodeIgniter4\system\Config\Services.php:664
 D:\projects\CodeIgniter4\system\Config\BaseService.php:258
 D:\projects\CodeIgniter4\system\Common.php:996
 D:\projects\CodeIgniter4\system\CodeIgniter.php:1075
 D:\projects\CodeIgniter4\system\CodeIgniter.php:549
 D:\projects\CodeIgniter4\system\CodeIgniter.php:361
 D:\projects\CodeIgniter4\tests\system\CodeIgniterTest.php:617
```

I don't know how to use `Session` in tests, but if you could help me I would appreciate it.

Reply:
@kenjis i remove the validations you mentioned and needed to update tests to work. i add the clause `@runTestsInSeparateProcesses` to the tests and they run in my machibe whitout errors (windows with apache php 7.4) but here they throw errors. Can you help me to see if how can i fix it please?

Reply:
fixed errors